examples/config.kinesis.reference.hocon

```diff
@@ -0,0 +1,167 @@
+    # "stdout" for writing good records to stdout
+    "type": "elasticsearch"
+
+    # Events are indexed using an Elasticsearch Client
+    # - endpoint: the cluster endpoint
+    # - port: the port the cluster can be accessed on
+    #   - for http this is usually 9200
+    #   - for transport this is usually 9300
+    # - username (optional, remove if not active): http basic auth username
+    # - password (optional, remove if not active): http basic auth password
+    # - shardDateFormat (optional, remove if not needed): formatting used for sharding good stream, i.e. _yyyy-MM-dd
+    # - shardDateField (optional, if not specified derived_tstamp is used): timestamp field for sharding good stream
+    # - maxTimeout (optional, default value 10000): the maximum attempt time before a client restart
+    # - maxRetries (optional, default value 6): the maximum number of request attempts before giving up
+    # - ssl: if using the http client, whether to use ssl or not
+    "client": {
+      "endpoint": "localhost"
+      "port": 9200
+      "username": "es-user"
+      "password": "es-pass"
+      "shardDateFormat": "_yyyy-MM-dd"
+      "shardDateField": "derived_tstamp"
+      "maxTimeout": 9999
+      "maxRetries": 5
+      "ssl": true
+    }
+
+    # When using the AWS ES service
+    # - signing: if using the http client and the AWS ES service you can sign your requests
+    #    http://docs.aws.amazon.com/general/latest/gr/signing_aws_api_requests.html
+    # - region where the AWS ES service is located
+    "aws": {
+      "signing": true
+      "region": "eu-central-1"
+    }
+
+    "cluster": {
+      # The Elasticsearch index name
+      "index": "good"
+      # The Elasticsearch index type.
+      # Index types are deprecated in ES >=7.x
+      # Therefore, it shouldn't be set with ES >=7.x
+      "documentType": "good-doc"
+    }
+
+    # Bulk request to Elasticsearch will be splitted to
+    # chunks according given limits.
+    # These values are optional.
+    "chunk": {
+      "byteLimit": 999999 # Default value is 1000000
+      "recordLimit": 499  # Default value is 500
+    }
+  }
+  "bad" {
+    # Bad sinks currently supported are:
+    # "kinesis" for writing bad records to Kinesis
+    # "stderr" for writing bad records to stderr
+    # "nsq" for writing bad records to NSQ
+    # "none" for ignoring bad records
+    "type": "kinesis"
+
+    # Stream name for events which are rejected by Elasticsearch
+    "streamName": "test-kinesis-bad-stream"
+
+    # Region where the Kinesis stream is located
+    "region": "eu-central-1"
+
+    # Optional endpoint url configuration to override aws kinesis endpoints,
+    # this can be used to specify local endpoints when using localstack
+    "customEndpoint": "127.0.0.1:7846"
+  }
+}
+
+# "good" for a stream of successfully enriched events
+# "bad" for a stream of bad events
+# "plain-json" for writing plain json
+"purpose": "good"
```

Comment:
I'd prefer `ENRICHED_EVENTS`, `SELF_DESCRIBING`, `JSON` as this is what they are in S3 Loader: https://github.com/snowplow/snowplow-s3-loader/blob/master/config/config.hocon.sample#L6-L8

For example, `bad` is really just self-describing JSONs, it (probably?) doesn't care if it's Snowplow-generated bad events or plain SDJSON.

Reply:
Ah sorry, I was planning to ask to you about this but I forgot to ask. 

I don't know all the details but we have some separate [BadEventTransformer](https://github.com/snowplow/snowplow-elasticsearch-loader/blob/6dec51cd00691b80a9fe7706749a9cb8828954f6/core/src/main/scala/com.snowplowanalytics.stream/loader/transformers/BadEventTransformer.scala). Do you think is it okay just convert it to `SELF_DESCRIBING`?

Reply:
You're right, there _is_ specific logic for Snowplow bad rows, it's not a simple SDJSON (in fact I wrote and then forgot about  it).

I'd then use `ENRICHED_EVENTS`, `BAD_ROWS`, and `JSON`. Eventually we can replace `BAD_ROWS` with `SELF_DESCRIBING`, but right now bad row schemas clash in several properties, so we cannot load them as is.